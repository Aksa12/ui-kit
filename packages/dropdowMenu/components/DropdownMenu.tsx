import * as React from "react";
import { cx } from "emotion";
import { menuList } from "../style";
import { border } from "../../shared/styles/styleUtils";

export interface DropdownMenuProps extends React.HTMLProps<HTMLDivElement> {
  children: React.ReactNode;
  maxHeight?: number;
  menuRef?: React.RefObject<HTMLDivElement>;
  width?: number;
}

const DropdownMenu = (props: DropdownMenuProps) => {
  const { maxHeight, menuRef, width, ...other } = props;
  return (
    <div
      className={cx(menuList, border("all"))}
      ref={menuRef}
      style={{ width, maxHeight }}
      data-cy="dropdownMenu"
      {...other}
    />
  );
};

export default DropdownMenu;
