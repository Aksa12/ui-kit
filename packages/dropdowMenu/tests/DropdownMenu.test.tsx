import React from "react";
import { shallow } from "enzyme";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import toJson from "enzyme-to-json";
import DropdownMenu from "../components/DropdownMenu";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("DropdownMenu", () => {
  it("renders", () => {
    const component = shallow(
      <DropdownMenu maxHeight={200} width={100}>
        menu content
      </DropdownMenu>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
