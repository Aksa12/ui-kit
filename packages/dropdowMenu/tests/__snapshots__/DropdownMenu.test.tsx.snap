// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu renders 1`] = `
.emotion-0 {
  background-color: #FFF;
  border-radius: 6px;
  box-shadow: 0 4px 8px rgba(0,0,0,0.1);
  overflow-y: scroll;
  border-width: 1px;
  border-color: #DADDE2;
  border-style: solid;
}

<div
  className="emotion-0"
  style={
    Object {
      "maxHeight": 200,
      "width": 100,
    }
  }
>
  menu content
</div>
`;
